subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/AbstractSnapshotWithChildrenTest.groovy

```diff
@@ -0,0 +1,208 @@
+
+    List<FileSystemNode> childrenWithSelectedChildReplacedBy(FileSystemNode replacement) {
+        children.collect { it == selectedChild ? replacement : it }
+    }
+
+    List<FileSystemNode> childrenWithAdditionalChild(FileSystemNode newChild) {
+        List<FileSystemNode> newChildren = new ArrayList<>(children)
+        newChildren.add(insertionPoint, newChild)
+        return newChildren
+    }
+
+    List<CHILD> childrenWithSelectedChildRemoved() {
+        children.findAll { it != selectedChild }
+    }
+
+    CHILD getNodeWithIndexOfSelectedChild(List<CHILD> newChildren) {
+        int index = children.indexOf(selectedChild)
+        return newChildren[index]
+    }
+
+    private int getInsertionPoint() {
+        int childIndex = SearchUtil.binarySearch(children) {
+            candidate -> PathUtil.compareFirstSegment(absolutePath, offset, candidate.pathToParent, CASE_SENSITIVE)
+        }
+        assert childIndex < 0
+        return -childIndex - 1
+    }
+
+    String getCommonPrefix() {
+        return selectedChild.pathToParent.substring(0, PathUtil.lengthOfCommonPrefix(selectedChild.pathToParent, absolutePath, offset, CASE_SENSITIVE))
+    }
+
+    String getPathFromCommonPrefix() {
+        return absolutePath.substring(offset + commonPrefix.length() + 1)
+    }
+
+    String getSelectedChildPathFromCommonPrefix() {
+        return selectedChild.pathToParent.substring(commonPrefix.length() + 1)
+    }
+
+    def getDescendantSnapshotOfSelectedChild(@Nullable MetadataSnapshot foundSnapshot) {
+        def descendantOffset = offset + selectedChild.pathToParent.length() + 1
+        1 * selectedChild.getSnapshot(absolutePath, descendantOffset, CASE_SENSITIVE) >> Optional.ofNullable(foundSnapshot)
+    }
+
+    def invalidateDescendantOfSelectedChild(@Nullable FileSystemNode invalidatedChild) {
+        def descendantOffset = offset + selectedChild.pathToParent.length() + 1
+        1 * selectedChild.invalidate(absolutePath, descendantOffset, CASE_SENSITIVE) >> Optional.ofNullable(invalidatedChild)
+    }
+
+    @SuppressWarnings("GrMethodMayBeStatic")
+    void noMoreInteractions() {
+        _ * _.pathToParent
+        0 * _
+    }
+
+    /**
+     * Different lists of relative paths of the initial children of the node under test.
+     */
+    static final INITIAL_CHILD_CONSTELLATIONS = [
+        ['name'],
+        ['name', 'name1'],
+        ['name', 'name1', 'name12'],
+        ['name', 'name1', 'name12', 'name2', 'name21'],
+        ['name/some'],
+        ['name/some/other'],
+        ['name/some', 'name2'],
+        ['name/some', 'name2/other'],
+        ['name/some', 'name2/other'],
+        ['name', 'name1/some', 'name2/other/third'],
+        ['aa/b1', 'ab/a1', 'name', 'name1/some', 'name2/other/third'],
+        ("a".."z").toList(),
+    ]
+
+    /**
+     * The queried/updated path has no common prefix with any of the initial children of the node under test.
+     *
+     * E.g.
+     *   path: 'name0/some
```

Comment:
This is the "selected path" right? Can we refer to it as so?

Reply:
Also maybe calling it something like "updated path" or `pathOfModification` would help?

Reply:
"selected path" = absolute path + offset

I would prefer to find a good name for this thing in https://github.com/gradle/gradle/pull/11378, given that it then corresponds to one variable.

Reply:
`pathOfModification` does work for `store` and `invalidate`, but not for `getSnapshot`.